### DISCUSSION TITLE
Welcome to Mixxx-Pioneer-DDJ-SB3 Discussions!

### DISCUSSION THREAD

**Discussion by julia-otran** [guilherme-otran/Mixxx-Pioneer-DDJ-SB3#2] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by iam-cult** [comment]
Hey, (been waiting to report this for a while), on the stable Mixxx 2.2.4, as soon as you load this preset it starts throwing errors. It's fine on 2.3 beta, but 2.3 is only available for a few distros. I looked at the error and it might have something to do with the 3rd and 4th decks, but I honestly have no Idea.

**Reply by julia-otran** [reply]
Hi, i think that maybe some installation problem. Did you copied all files that are listed in the readme.md?

Im just using Mixxx 2.2.4 on ubuntu and it is working fine.

Please, post the log information so I can help you better (screen shots, also remember to click `Show Details` on the error dialog so i could see the log.

**Comment by iam-cult** [comment]
Any thoughts on merging with the upstream repository? It seems a little strange to be operating out of a fork. I'm sure the original creator would also appreciate a merge.

**Reply by julia-otran** [reply]
I've opened https://github.com/Dancephy/Mixxx-Pioneer-DDJ-SB3/pull/8. However, there are lot of changes. Anyway, let's see what happens

**Comment by mhaddestdawg** [comment]
Hi, I currently have your preset and I am having a hard time getting decks 3 & 4 to work they aren't lit up and pressing them doesn't do anything either.
Best,
Andrew

**Reply by julia-otran** [reply]
Well, my controller wasn't working deck 3/4 because the BPM sysex message was not sent. After sending, it starts to work.

You might want to run mixxx with --controllerDebug flag and check if there is a sysex message beign sent to the controller when you press deck3/4 buttons (also, you may check that changing tempo will also send the BPM sysex message). If it is not being sent, then you may have not installed correctly the preset. I will recommend that you run the script and choose yes to overriding your preset.

Maybe you have got another revision for the controller and the message may be different. If that is the case, you may attach a usb debug tool to see the structure of the message that is being sent to the controller when using serato.

**Comment by mhaddestdawg** [comment]
I got it to work after that, thanks also I was meaning to ask is there a way to use the beat jump features with the SB3 in your mapping?
Best,
Andrew

**Reply by julia-otran** [reply]
well beat-jump should work:

Suppose that this is the pads:
```
+---+---+---+---+
| a | b | c | d |
+---+---+---+---+
| e | f | g | h |
+---+---+---+---+
```

a: beat-jump halve
b: beat-jump double
c: go back
d: go forward

the other buttons will go back and go front according to a pre-determinated size. You also can use shift for it

**Comment by mhaddestdawg** [comment]
Awesome thanks I’m kinda green in terms of DJ controllers since I’m still getting used to the SB3 your help and quick response helps a lot. Thanks
